src/openvic-simulation/map/Province.cpp

```diff
@@ -137,6 +137,18 @@ bool Province::add_adjacency(Province const* province, distance_t distance, flag
 	return true;
 }
 
+fvec2_t Province::get_unit_position() const {
+	return positions.unit.value_or(positions.center);
+}
+
+Province::distance_t Province::calculate_distance_to(Province const* province) const {
+	const fvec2_t my_unit_position = get_unit_position();
+	const fvec2_t other_unit_position = province->get_unit_position();
+	const fvec2_t distance_vector = other_unit_position - my_unit_position;
+	const fixed_point_t distance = distance_vector.length_squared();
+	return static_cast<Province::distance_t>(distance.to_int64_t());
```

Comment:
Pleas change the `distance_t` alias in `Province.hpp` to `fixed_point_t`, and then remove the cast and `to_int64_t` from here.